src/message.rs

```diff
@@ -9,6 +9,41 @@ use std::marker::PhantomData;
 use std::mem;
 use tibrv_sys::*;
 
+pub struct MsgIter<'m> {
+    msg: &'m Msg,
+    index: u32,
+}
+
+impl<'m> Iterator for MsgIter<'m> {
+    type Item = Result<BorrowedMsgField<'m>, TibrvError>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let num_fields = match self.msg.num_fields() {
+            Ok(n) => n,
+            Err(e) => return Some(Err(e)),
+        };
+        let r = if self.index >= num_fields {
+            None
+        } else {
+            Some(self.msg.get_field_by_index(self.index))
+        };
+        self.index += 1;
+        r
+    }
+}
+
+impl<'m> IntoIterator for &'m Msg {
+    type IntoIter = MsgIter<'m>;
+    type Item = <Self::IntoIter as Iterator>::Item;
+
+    fn into_iter(self) -> Self::IntoIter {
+        MsgIter {
+            msg: self,
+            index: 0,
+        }
+    }
+}
+
```

Comment:
> Not sure if implementing this on `&Msg` is the right general solution, or how it interacts with `BorrowedMsg`.

Yep this looks good,, `BorrowedMsg` only exists to represent a nested `Msg` in a field. At the moment basically nothing is implemented for these nested messages, you need to call `.to_owned()` on it and obtain a `Msg` before you can do useful stuff like iterate over the fields. This could probably be improved.

Reply:
Looks like I made a mess of using the review tool!